addons/mail/static/src/messaging/component/notification_list/notification_list_notification_group_tests.js

```diff
@@ -0,0 +1,575 @@
+odoo.define('mail.messaging.component.NotificationList_NotificationGroupTests', function (require) {
+'use strict';
+
+const components = {
+    NotificationList: require('mail.messaging.component.NotificationList'),
+};
+
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('NotificationList_NotificationGroup', {
+    beforeEach() {
+        utilsBeforeEach(this);
+
+        /**
+         * @param {Object} param0
+         * @param {string} [param0.filter='all']
+         */
+        this.createNotificationListComponent = async ({ filter = 'all' } = {}) => {
+            const NotificationListComponent = components.NotificationList;
+            NotificationListComponent.env = this.env;
+            this.component = new NotificationListComponent(null, { filter });
+            await afterNextRender(async () => {
+                await this.component.mount(this.widget.el);
+            });
```

Comment:
inline? Or at least "inline" (=non-scoped) arrow function

Reply:
I thought that when we did an async function that we didn't care about the return value it should do await inside and not return the result?

Reply:
indeed, it prevents some potential programming errors

Reply:
So I can ignore your comment here and keep it as it is?

Reply:
alright